draft-ietf-core-oscore-groupcomm.md

```diff
@@ -1278,21 +1300,21 @@ Upon receiving a protected response with the Group Flag set to 0, following the
+This section describes how a server endpoint can verify freshness of a request by means of a challenge-response exchange with a client using the Echo Option for CoAP specified in {{Section 2 of RFC9175}}. The same mechanism, with small alterations, is also used by the server when first processing a request using a Recipient Context whose Replay Window was initialized as invalid.
 
-If the application requires freshness, e.g., according to time- or event-based policies (see {{Section 2.5.1 of RFC9175}}), a server proceeds as described below, upon receiving a request from a particular client for the first time.
+If the application requires freshness, e.g., according to time- or event-based policies (see {{Section 2.5.1 of RFC9175}}), a server proceeds as described below upon receiving a request from a particular client for the first time.
 
 The server processes the message as described in this document, but, even if valid, does not deliver it to the application. Instead, the server replies to the client with a Group OSCORE protected 4.01 (Unauthorized) response message, including only the Echo Option and no diagnostic payload. The server MUST use its Partial IV when generating the nonce for protecting the response conveying the Echo Option, and MUST include the Partial IV in the response.
 
 The Echo Option value SHOULD NOT be reused; if it is reused, it MUST be highly unlikely to have been recently used with this client. Since this response is protected with the Security Context used in the group, the client will consider the response valid upon successfully decrypting and verifying it.
 
 The server stores the Echo Option value included in the response together with the pair (gid,kid), where 'gid' is the Group Identifier of the OSCORE group and 'kid' is the Sender ID of the client in the group. These are specified in the 'kid context' and 'kid' fields of the OSCORE Option of the request, respectively. After a group rekeying has been completed and a new Security Context has been established in the group, which results also in a new Group Identifier (see {{sec-group-key-management}}), the server MUST delete all the stored Echo values associated with members of the group.
 
-Upon receiving a 4.01 (Unauthorized) response that includes an Echo Option and originates from a verified group member, the subsequent client request echoing the Echo Option value MUST be sent as a unicast message to the same server.
+After receiving a 4.01 (Unauthorized) response that includes an Echo Option and originates from a verified group member, a subsequently sent client request echoing the Echo Option value MUST be sent as a unicast message to the same server.
```

Comment:
```suggestion
After receiving a 4.01 (Unauthorized) response that includes an Echo Option and originates from a verified group member, the next client request MUST echo the Echo Option value and MUST be sent as a unicast message to that server.
```

Reply:
After taking the suggestion into account, we thought about a different update that makes the new text clearer while still reflecting the intended behavior specified in Sections 2.3 and 2.4 of RFC 9175 (more details will be in a follow-up mail).

The updated new text reads as below. There will a commit for that.

> After receiving a 4.01 (Unauthorized) response that includes an Echo Option and originates from a verified group member, **a subsequent client request sent to that server and echoing the Echo Option value MUST be sent as a unicast message to that server.**

Reply:
Fixed in https://github.com/core-wg/oscore-groupcomm/pull/114/commits/3f0f2e71d8d848a126a5d79fe2944471f3e62842